service.rb

```diff
@@ -7,49 +7,42 @@ class Service
   def initialize(port)
     app = -> environment {
       request = Rack::Request.new(environment)
+      response = Rack::Response.new
       store = YAML::Store.new("daily_data.yml")
 
       if request.get? && request.path == "/show/data"
-        content_type = "text/html"
-        status = 200
-        response_message = "<ul>\n"
+        response.write("<ul>\n")
+        response.content_type = "text/html; charset=UTF-8"
```

Comment:
Extra credit: it’s kind of annoying to have to set the same sensible default `charset` on every `Content-Type`; is there a way you could move this responsibility [somewhere else](https://github.com/adamluzsi/rack-charset)?